code/modules/rykka_puzzle_wips/reactor_defense.dm

```diff
@@ -0,0 +1,795 @@
+ *	preferably undamaged through a number of waves, and you're able to progress. Mechanical challenges are more fun than "shoot thing, win". More to come.	//
+ *////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/*												Object Definition, Initialize, Process															*/
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+GLOBAL_LIST_EMPTY(reactor_mob_spawners) // Define our global list here. This is used for the spawners.
+
+// List of tabs used for TGUI Displays and Reactor Stage Processing. If you're not sure how defines work, refer to the .dm reference docs.
+#define IDLE 1
+#define WARMUP 2
+#define ENGAGED 3
+#define FINISHED 4
+#define VERIFYING 5
+
+/obj/structure/damaged_reactor // Damaged Reactor object used as the "thing" we interact with
+	name = "damaged reactor"
+	desc = "You shouldn't be able to read this! OOC: Poke @Rykka Stormheart or ask in dev, the reactors should be a subtype of damaged_reactor!"
+	icon = 'icons/obj/damaged_reactor.dmi'
+	icon_state = "reactor_off"
+	density = 1
+	anchored = TRUE
+	climbable = FALSE
+	breakable = FALSE
+	var/icon_state_warmup = null					// Icon to use when the reactor is engaged/warming up. TODO: when I get sprites.
+	var/icon_state_severe_damage = null				// Icon to use when the reactor has taken severe damage. TODO: when I get sprites.
+	var/icon_state_powered = null					// Icon to use once the reactor is at full power. TODO: when I get sprites.
+	var/active_defense = FALSE						// Are we actively in "defense" mode? (IE has someone engaged the defense gamemode?) This variable governs if the reactor can take damage or not (FALSE = immune!)
+	var/list/spawners = list()						// The list of wave spawners we have (Defined at init.)
+	var/reactor_id = null							// Our ID, used to link the reactor + the spawners specific to it, as well as the control console. This MUST be a WORD, not a NUMBER.
+	var/state = IDLE								// What state are we in currently? (This is used for tracking what step we're on.)
+	var/verification_successful = FALSE				// This is false by default, do NOT change it to true manually. Procs will handle this.
+	
+	// Customization Options. Use these to customize how you want the mode to play out.
+	var/health = 2000								// The reactor's health value (Set this to be what maxhealth is, the code will handle the rest).
+	var/maxhealth = 2000							// The maximum health this object can have.
+	var/list/waves = list(1)						// Total amount of waves we'll have, and what mob lists we'll use PER wave. Add more entries to the list to increase the # of waves.

```

Comment:
IIRC the list is to allow you to do list(1,2,1) and the wave_mobs will according spawn from list 1, list 2, list 1 on wave 1/2/3 respectively.

Leaving this up so it can be discussed further.